server/src/main/java/com/linecorp/centraldogma/server/internal/storage/DirectoryBasedStorageManager.java

```diff
@@ -323,6 +306,68 @@ public T unremove(String name) {
         return unremovedChild;
     }
 
+    @Override
+    public void purge(String name) {
+        ensureOpen();
+        validateChildName(name);
+        final File removed = new File(rootDir, name + SUFFIX_REMOVED);
+        final File purged = new File(rootDir, name + SUFFIX_PURGED);
+
+        if (removed.exists() && !purged.exists()) { // try to rename `.removed` into `.purged`
```

Comment:
- Could reduce the number of system calls:
  ```java
  if (!purged.exists()) {
     if (removed.exists()) {
         ...
     } else {
         ...
     }
  }
  ```
- Shouldn't we also raise `newStorageNotFoundException(...)` when only `.purged` exists?

Reply:
Ahh, reduce system calls!! I didn't think about that. It's a nice idea. Thanks 🙏